### PR TITLE
add JWKS_RESOLVER_INSECURE_SKIP_VERIFY flag

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

without this PR, it's painful to retrive from a local keycloak with self-signed cert.

### CODE REVIEW

File: pilot/pkg/model/jwks_resolver.go [hunk 1]
```diff
@@ -212,8 +212,10 @@ func newJwksResolverWithCABundlePaths(
 				Proxy:             http.ProxyFromEnvironment,
 				DisableKeepAlives: true,
 				TLSClientConfig: &tls.Config{
-					RootCAs:    caCertPool,
-					MinVersion: tls.VersionTLS12,
+					// nolint: gosec
```
Comment (howardjohn) [inline pilot/pkg/model/jwks_resolver.go:215]: nit: any nolint:gosec should explain _why_. In this case its simple - user explicitly opted into insecure.

**Review by howardjohn — COMMENTED** [review]
One note: with JwksFetchMode you can move fetching to Envoy which uses standard Istio config (DR) to configure TLS which can already do this. 

cc @liwenhao0810 @aryan16 for more review

**Comment by liwenhao0810** [comment]
> One note: with JwksFetchMode you can move fetching to Envoy which uses standard Istio config (DR) to configure TLS which can already do this.
> 
> cc @liwenhao0810 @aryan16 for more review

Yes when set JwksFetchMode to envoy to delegate fetching to envoy, and with the DR settings tls simple without any cert, envoy's default is to skip verify. 
While if you want istiod fetching, this make sense to me.

**Comment by zirain** [comment]
> > One note: with JwksFetchMode you can move fetching to Envoy which uses standard Istio config (DR) to configure TLS which can already do this.
> > cc @liwenhao0810 @aryan16 for more review
> 
> Yes when set JwksFetchMode to envoy to delegate fetching to envoy, and with the DR settings tls simple without any cert, envoy's default is to skip verify. While if you want istiod fetching, this make sense to me.

TIL

BTW, what's benifit of using istiod as default?

**Comment by ramaraochavali** [comment]
implementation to fetch JWT by Istiod was implemented when Envoy did not have the ability to fetch it by it self using remote uri. Now that since Envoy has it with caching etc - I can not think of a case where using Istiod has benefits except that Istiod fetches it once for all proxies where it is needed instead of every proxy doing it

**Comment by zirain** [comment]
@ramaraochavali tanks for clarify this, is it safe to switch envoy way as default?

**Comment by ramaraochavali** [comment]
> is it safe to switch envoy way as default?

It is safe unless @howardjohn  @hzxuzhonghu have any other concerns.

**Comment by hzxuzhonghu** [comment]
Switching to envoy mode can be unsafe, this requires a service exist.

>Envoy has it with caching etc 

If envoy can be able to access jwks, fetching via envoy can benefit more

**Comment by ramaraochavali** [comment]
>Switching to envoy mode can be unsafe, this requires a service exist.

We can create that cluster via Bootstrap similar to how we do for access log and metrics service

**Comment by hzxuzhonghu** [comment]
I mean this may require user additional config, it is not a smooth upgrade

**Comment by howardjohn** [comment]
If we had done it from the beginning we could safely move. Changing now by
default transparently is pretty hard.

**Comment by zirain** [comment]
@howardjohn so I think this still worth to merge.

File: pilot/pkg/features/pilot.go [hunk 2]
```diff
@@ -663,6 +663,9 @@ var (
 		"Metric scope rotation interval, set to 0 to disable the metric scope rotation").Get()
 	MetricGracefulDeletionInterval = env.Register("METRIC_GRACEFUL_DELETION_INTERVAL", 5*time.Minute,
 		"Metric expiry graceful deletion interval. No-op if METRIC_ROTATION_INTERVAL is disabled.").Get()
+
+	JwksResolverInsecureSkipVerify = env.Register("JWKS_RESOLVER_INSECURE_SKIP_VERIFY", false,
+		"If enabled, the JWKS resolver will skip verifying the certificate of the JWKS server.").Get()
```
Comment (hzxuzhonghu) [inline pilot/pkg/features/pilot.go:668]: istiod will skip verifying

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM for dev usage. But we should use ca cert mount for production with `jwksResolverExtraRootCA` set

**Review by howardjohn — APPROVED** [review]

**Pull request merged**